### ISSUE TITLE
@ctix-declaration is too strict.

### ISSUE THREAD

**Issue by smallmain** [imjuni/ctix#145]
`singleton-check.ts`
```ts
/** @ctix-declaration */
/* eslint-disable no-var -- global neeed this. */
import { VERSION } from "internal/constants";
import { internal } from "./internal.js";

if (globalThis["XENON_INSTANCE_VERSION"] !== undefined) {
    internal.log.warn(
        "ATTENTION!!!",
        "There may currently be multiple versions of the module.",
        "Version A:",
        VERSION,
        "Version B:",
        globalThis["XENON_INSTANCE_VERSION"],
    );
} else {
    globalThis["XENON_INSTANCE_VERSION"] = VERSION;
}

declare global {
    var XENON_INSTANCE_VERSION: string | undefined;
}
```

 I used the above file to check for multiple module instances, which requires an import statement in 'index.ts', but that would be excluded.

It is expected that files with '@ctix-declaration' will simply generate an 'import' statement in all cases.

**Comment by imjuni** [comment]
@smallmain Hi,

The `@ctix-declaration` comment is intended for use in cases where the `declare module` syntax cannot be added to the index.ts file, rather than as an exception process for the exclude option. The order and process of including or excluding target files can significantly impact the ctix user, so in this case, it is recommended to use the exclude option more precisely.

**Comment by smallmain** [comment]
> @smallmain Hi,
> 
> The `@ctix-declaration` comment is intended for use in cases where the `declare module` syntax cannot be added to the index.ts file, rather than as an exception process for the exclude option. The order and process of including or excluding target files can significantly impact the ctix user, so in this case, it is recommended to use the exclude option more precisely.

Hi, can you tell me how to solve this situation?

Seems to be missing a feature that is always included.

**Comment by imjuni** [comment]
@smallmain Could you please provide a reproducible repository?

**Comment by smallmain** [comment]
> @smallmain Could you please provide a reproducible repository?

@imjuni 

[ctix-demo.zip](https://github.com/user-attachments/files/16577103/ctix-demo.zip)

Please take a look at the archive for the minimal reproduction project.

I need to generate an import statement to 'check.ts' in' index.ts', but there's no way to do that right now.

**Comment by imjuni** [comment]
@smallmain 

I developed `@ctix-declaration` specifically for extending types, so it is designed not to work in files that contain if/else or for/while, etc statements. In the reproducible repo you provided, `@ctix-declaration` did not work because of the if/else statements.

While TypeScript/JavaScript supports extending or overwriting `globalThis`, and therefore ctix should also support it, ctix is a code generation tool and must ensure that the generated code is always compilable. For this reason, ctix supports extending or overwriting `globalThis`, but it does not support having export statements in the same file. In such cases, it becomes difficult to predict how the correct import/export code should be generated for a valid compilation.

I have released version 2.6.1, which includes these updates. Thank you for your continuous support in helping improve ctix!

**Comment by smallmain** [comment]
> @smallmain
> 
> I developed `@ctix-declaration` specifically for extending types, so it is designed not to work in files that contain if/else or for/while, etc statements. In the reproducible repo you provided, `@ctix-declaration` did not work because of the if/else statements.
> 
> While TypeScript/JavaScript supports extending or overwriting `globalThis`, and therefore ctix should also support it, ctix is a code generation tool and must ensure that the generated code is always compilable. For this reason, ctix supports extending or overwriting `globalThis`, but it does not support having export statements in the same file. In such cases, it becomes difficult to predict how the correct import/export code should be generated for a valid compilation.
> 
> I have released version 2.6.1, which includes these updates. Thank you for your continuous support in helping improve ctix!

Thank you, we just need a way to include the file anyway and generate 'import "xxx"; 'directive, this directive can be guaranteed to be valid by the user.
